vdom/core.py

```diff
@@ -170,6 +170,8 @@ class Component():
         """A basic class for a virtual DOM Component"""
 
         def __init__(self, *children, **kwargs):
+            if not allow_children and children:
+                raise ValueError('<{tagName} /> cannot have children'.format(tagName=tagName))
             self.children = _flatten_children(*children, **kwargs)
```

Comment:
I realize now after working with this that the `children` check should come after `_flatten_children` since it can come in as a kwarg.